Mathlib/Topology/IndicatorConstPointwise.lean

```diff
@@ -0,0 +1,73 @@
+/-
+Copyright (c) 2023 Kalle Kytölä. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kalle Kytölä
+-/
+import Mathlib.Algebra.IndicatorFunction
+import Mathlib.Topology.Separation
+
+/-!
+# Pointwise convergence of indicator functions
+
+In this file, we prove the equivalence of three different ways to phrase that the indicator
+functions of sets converge pointwise.
+-/
+
+
+open Filter Topology
+
+variable {α : Type*} {A : Set α}
+variable {ι : Type*} (L : Filter ι) [IsCountablyGenerated L] {As : ι → Set α}
+
+lemma tendsto_indicator_const_iff_forall_eventually' {β : Type*} [Zero β] [TopologicalSpace β]
+    (b : β) {B O : Set β} (B_nhd : B ∈ 𝓝 b) (nin_B : 0 ∉ B) (O_nhd : O ∈ 𝓝 0) (nin_O : b ∉ O) :
```

Comment:
Since `B` and `O` do not appear in the conclusion, I would just replace the assumptions by `{0}ᶜ ∈ 𝓝 b` and `{b}ᶜ ∈ 𝓝 0`.

Reply:
I think this is the heart of this lemma:
```lean
lemma tendsto_ite {β : Type*} {p : ι → Prop} [DecidablePred p] {q : Prop} [Decidable q]
    {a b : β} {F G : Filter β} (h1 : {a}ᶜ ∈ G) (h1 : {b}ᶜ ∈ F) :
    Tendsto (fun i ↦ if p i then a else b) L (if q then F else G) ↔ 
      ∀ᶠ i in L, p i ↔ q := by sorry
```
That might be nice to add, and your version should only be a few lines afterwards.

This version also doesn't use topology.

(There is probably also a version where `a` and `b` may depend on `i`, and where `{a}` is then replaced by `range a`, but that might be overkill).

Reply:
I could only prove `tendsto_ite` after adding the hypotheses `(haF : principal {a} ≤ F)` and ` (hbG : principal {b} ≤ G)`.